public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -494,6 +478,18 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
     return merge(...subQueries);
   }
 
+  isTraceQlMetricsQuery(query: string): boolean {
+    const metricsFnRegex =
+      /\|\s*(rate|count_over_time|avg_over_time|max_over_time|min_over_time|quantile_over_time)\s*\(/;
+    return !!query.trim().match(metricsFnRegex);
+  }
```

Comment:
> Detect when the query contains a metrics function

Maybe we can add this as a code comment. `isTraceIdQuery` already has one, so it could make sense also for `isTraceQlMetricsQuery`.